Mathlib/Data/Setoid/Basic.lean

```diff
@@ -196,6 +196,14 @@ theorem eq_top_iff {s : Setoid α} : s = (⊤ : Setoid α) ↔ ∀ x y : α, s x
   rw [_root_.eq_top_iff, Setoid.le_def, Setoid.top_def]
   simp only [Pi.top_apply, Prop.top_eq_true, forall_true_left]
 
+instance : Subsingleton (Quotient (⊤ : Setoid α)) where
+  allEq x y := Quotient.inductionOn₂ x y fun a b ↦ by simp
+
+instance {s : Setoid α} [hs : Fact (s ≠ ⊤)] : Nontrivial (Quotient s) := by
+  refine not_subsingleton_iff_nontrivial.mp fun H ↦ hs.out ?_
+  simp_rw [eq_top_iff, ← Quotient.eq]
+  exact fun _ _ ↦ H.allEq _ _
```

Comment:
```suggestion
  rw [← not_subsingleton_iff_nontrivial, Quotient.subsingleton_iff]
  exact hs.out
```

Reply:
Probably we should have `nontrivial_iff` too

Reply:
I added `subsingleton_iff` instead, I think people can do a few rewrites if they need `nontrivial_iff`.